File: Arrays/Search Insert Position.cpp
class Solution {
public:
    int searchInsert(vector<int>& nums, int target) {
        if(nums[0] > target)
            return 0;
        int start = 0, end = nums.size()-1, mid;
        while(start <= end){
            mid = (start + end) / 2;
            if(nums[mid] == target)
                return mid;
            else if(nums[mid] < target)
                start = mid + 1;
            else
                end = mid - 1;
            }
        return start;
    }
};

File: Arrays/Search Insert Position.rs
impl Solution {
    pub fn search_insert(nums: Vec<i32>, target: i32) -> i32 {
        let mut left = 0;
        let mut right = nums.len()-1;

        while left <= right && right < nums.len() {
            let mid = (left + right) / 2;

            if nums[mid] == target {
                return mid as i32;
            }
            else if nums[mid] > target {
                right = mid - 1;
            } else {
                left = mid + 1;
            }
        }
        left as i32
    }
}